src/auth/index.js

```diff
@@ -26,12 +37,33 @@ export default class Auth {
       storageType: 'localStorage'
     }, options);
 
-    this.client = client;
-    this.rootUrl = rootUrl;
-    this.codec = options.codec;
-    this.platform = options.platform || _platform;
-    this.storage = createStorage(options);
-    this.providers = createProviders(this, options);
+    auth.client = client;
+    auth.rootUrl = rootUrl;
+    auth.codec = options.codec;
+    auth.platform = options.platform || _platform;
+    auth.storage = createStorage(options);
+    auth.providers = createProviders(auth, options);
+
+    Promise.all([
+      auth._get(),
+      auth.storage.get(authCommon.REFRESH_TOKEN_KEY),
+      auth.storage.get(authCommon.USER_LOGGED_IN_PT_KEY),
+      auth.storage.get(authCommon.DEVICE_ID_KEY)
+    ]).then(([authObj, rt, loggedInProviderType, deviceId]) => {
```

Comment:
[q] If the clients rely on this resolving, are we assuming that a reject from storage always means there was a bad failure. In addition to that, does that mean we assume the contract is if you get something that isn't set as a key, it returned undefined/null? If yes to both of those, then this is fine as is; otherwise, we'd need to handle the reject case in the clients to continue on.

Reply:
So how do we handle the rejection case here then?